pkg/workloads/console/controller.go

```diff
@@ -149,8 +176,43 @@ func (r *ConsoleReconciler) Reconcile(request reconcile.Request) (res reconcile.
 		return res, err
 	}
 
+	// Update the status fields in case they're out of sync, or the console spec
+	// has been updated
+	csl, err = r.updateStatus(csl, job)
+	if err != nil {
+		return res, err
+	}
+
+	// Terminate if necessary
+	if isJobExpired(csl) {
```

Comment:
I think this is going to try to delete a job multiple times if, for example, a console expires but then its template is updated. Is that worth worrying about?

Reply:
This wasn't actually happening because a little further up there's an early return which bails out if the job is not found but the console expires.

This is bad though I think - it prevents `updateRoleBindings` and `updateStatus` from being called. So I'm going to rework this.